src/wms/src/test/java/org/geoserver/wms/legendgraphic/BufferedImageLegendGraphicOutputFormatTest.java

```diff
@@ -252,15 +260,58 @@ public void testMultipleLayers() throws Exception {
 
         assertPixel(image, 10, 50 + titleHeight, new Color(224, 64, 0));
 
-        // second title
-        assertPixel(image, 1, 60 + titleHeight + titleHeight / 2, new Color(0, 0, 0));
-
         // same colors for the second layer
         assertPixel(image, 10, 70 + titleHeight * 2, new Color(192, 160, 0));
 
         assertPixel(image, 10, 90 + titleHeight * 2, new Color(0, 0, 0));
 
         assertPixel(image, 10, 110 + titleHeight * 2, new Color(224, 64, 0));
+
+        // 128 * 3 = search for pixels are at least half black  (3 = number of components - RGB)
+        assertManyPixelsOfColor(image, Color.BLACK, 128 * 3, true, 1500);
+    }
+
+    /**
+     * Looks through the image and counts the number of pixels that are approximately the given
+     * color. This must be greater than minNumber.
+     *
+     * @param image image to look through
+     * @param color test color
+     * @param tolerance how close the pixel's test color needs to be to be a match (0=exact match)
+     * @paaram onlyGrey only consider grey-scale pixels
+     * @param minNumber how many pixels in the image must be the given color
+     */
+    protected void assertManyPixelsOfColor(
+            BufferedImage image, Color color, int tolerance, boolean onlyGrey, int minNumber) {
```

Comment:
Idea: If we are going to see a lot of these kind of comparisons (hopefully not) it would be useful to have this comparison as an option for ImageAssert (that way we have a workflow to review and manage reference images).

Reply:
I don't think we can do image references for this - the images (and fonts) are different sizes on different machines.  This leads to quite different comparisons.